R/test_description.R

```diff
@@ -187,6 +187,20 @@ srcref_str <- function(ref) {
 #' @param ref a \code{srcref}
 #'
 expr_str <- function(ref) {
+  # special case naked generic calls (see: inst/examplepkg/tests/non-testthat.R)
+  if (inherits(ref[[1]], "standardGeneric")) {
+    generic <- ref[[1]]@generic
+    arg_types <- vapply(ref[-1], function(i) class(i)[[1]], character(1L))
+
+    str <- paste0(
+      "S4 Generic Call: ", 
+      generic, 
+      "(", paste0("<", arg_types, ">", collapse = ", "), ")"
+    )
+
+    return(str)
+  }
+
   if (is.expression(ref)) ref <- as.call(ref)
   gsub("\\s{2,}", " ", paste0(deparse(ref), collapse = "; "))
 }
```

Comment:
This is the crux of the bug fix. Previously, trying to deparse a "standardGeneric" object would throw an error.

Now we first check whether the call is to a standard generic and write out a description of the call and object types that invoked the method.